### PR TITLE
Validate unterminated comments

### LINKED ISSUE #18: Parser does not validate unterminated comments
An XPath expression ending with an unterminated constant should result in error [XPST0003]. Currently these expressions are accepted by the parser.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #18

### CODE REVIEW

**Comment by sharwell** [comment]
|  | Tests Run | Failures | Errors | Skipped |
| --- | --- | --- | --- | --- |
| Before | 13414 | 644 | 226 | 0 |
| After | 13414 | 641 | 226 | 0 |

File: bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/XPathFlex.java [hunk 1]
```diff
@@ -853,6 +854,10 @@ private void zzDoEOF() throws java.io.IOException {
       zzEOFDone = true;
       yyclose();
     }
+
+	if (commentLevel > 0) {
+		throw new XPathParserException("Unterminated comment", null);
+	}
```
Comment (RackerWilliams) [inline bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/XPathFlex.java:862]: This is generated code, the modification should probably be placed in the grammar file...under `grammars/xpath.lex`

``` flex
  %eof{
        ...
   %eof}
```

Comment (sharwell) [inline bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/XPathFlex.java:862]: Done!

**Comment by RackerWilliams** [comment]
Looks good 👍

**Pull request merged**